addons/web_editor/models/ir_ui_view.py

```diff
@@ -80,10 +84,76 @@ def save_oe_structure(self, el):
             'mode': 'extension',
         }
         vals.update(self._save_oe_structure_hook())
-        self.env['ir.ui.view'].create(vals)
+        oe_structure_view = self.env['ir.ui.view'].create(vals)
+        self._copy_custom_snippet_translations(oe_structure_view, 'arch_db')
 
         return True
 
+    @api.model
+    def _copy_custom_snippet_translations(self, record, html_field):
+        """ Given a ``record`` and its HTML ``field``, detect any
+        usage of a custom snippet and copy its translations.
+        """
+        lang_value = record[html_field]
+        if not lang_value:
+            return
+
+        tree = html.fromstring(lang_value)
+        for custom_snippet_el in tree.xpath('//*[hasclass("s_custom_snippet")]'):
+            custom_snippet_name = custom_snippet_el.get('data-name')
+            custom_snippet_view = self.search([('name', '=', custom_snippet_name)], limit=1)
+            if custom_snippet_view:
+                self._copy_model_field_terms_translations(custom_snippet_view, 'arch_db', record, html_field)
+
+    @api.model
+    def _copy_model_field_terms_translations(self, record_from, name_field_from, record_to, name_field_to):
```

Comment:
I still think the name can be improved
In ORM we have terminologies like `model translated field` and `model_terms translated field`.
So I suggested `_copy_field_model_terms_translations`
But in `_copy_model_field_terms_translations`, the `model` contains no information because `field` is always for `model`
If you want to keep it as short as possibe, maybe `_copy_field_terms_translations` or `_copy_field_translated_terms`

Reply:
> So I suggested _copy_field_model_terms_translations
> [..]
> If you want to keep it as short as possibe [..]

It's not about short or not, it's just that I can't wrap my head around what is a "field model terms translations", maybe it's just me tho, but it really bugs me out.

It's why I reverted the terms, "model field terms translations" is really easy to understand IMO: it copies the given _model's field's terms' translations_.

I'll go with your `_copy_field_terms_translations` suggestion,